app/src/main/java/ch/epfl/sdp/blindly/profile_setup/ProfileAudioRecording.kt

```diff
@@ -6,18 +6,35 @@ import android.view.View
 import androidx.appcompat.app.AppCompatActivity
 import ch.epfl.sdp.blindly.R
 import ch.epfl.sdp.blindly.recording.RecordingActivity
+import ch.epfl.sdp.blindly.user.User
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
 
 /**
  * Page to prompt user to record an audio file for their profile.
  */
 class ProfileAudioRecording : AppCompatActivity() {
+
+    private lateinit var userBuilder: User.Builder
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.profile_setup_audio_recording)
+
+        val bundle = intent.extras
+        if (bundle != null) {
+            userBuilder = bundle.getString(EXTRA_USER)?.let { Json.decodeFromString(it) }!!
```

Comment:
Is it useful to get the userBuilder if you don't set anything more to it in this activity? Would it not be possible to just put the val bundle(or rather a lateinit var bundle) in the intent.putExtras(bundle) ?

Reply:
you're right !